% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/input.R
\name{get_oncoscan_coverage_from_probes}
\alias{get_oncoscan_coverage_from_probes}
\title{Load the oncoscan annotation file and infer the covered regions from it.}
\usage{
get_oncoscan_coverage_from_probes(filename = system.file("extdata",
  "OncoScan.na33.r1.annot.csv.zip", package = "oncoscanR"))
}
\arguments{
\item{filename}{Path to the ChAS annotation file (either compressed or not). Defaults to the Oncoscan na33.r1
file present in the package.}
}
\value{
A \code{GRanges} object containing the regions covered on each chromosome arm. If the file does
not respect the format specifications, then an error is raised.
}
\description{
Load the oncoscan annotation file and infer the covered regions from it.
}
\details{
Expects the following columns from the annotation file (as the first line of non-comment):
  - "Chromosome": chromosome name
  - "Physical Position": genomic position of the SNP
  - "Cytoband": cytoband name (e.g. 3q22.1)
}
\examples{
oncoscan_na33.cov <- get_oncoscan_coverage_from_probes()
}
